beacon-chain/core/state/interop_regression_test.go

```diff
@@ -55,7 +55,68 @@ func TestTopazRewardsRegression_StateRootValid(t *testing.T) {
 	// expect state transition to fail, as the block had an incorrect state root
 	postStateObj, err := ExecuteStateTransition(context.Background(), stateObj, blk)
 	if err == nil {
-		t.Error(err)
+		t.Error("State transition did not fail")
+	}
+	expectedRt, err := expectedStateObj.HashTreeRoot(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	rt, err := postStateObj.HashTreeRoot(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	if expectedRt != rt {
+		t.Errorf("Wanted a state root of %#x but got %#x", expectedRt, rt)
+	}
+	if !ssz.DeepEqual(expectedStateObj.InnerStateUnsafe(), postStateObj.InnerStateUnsafe()) {
+		diff, _ := messagediff.PrettyDiff(expectedStateObj.InnerStateUnsafe(), postStateObj.InnerStateUnsafe())
+		t.Errorf("Difference between states: %s", diff)
+	}
+}
+
+func TestSchlesiRewardsRegression_PyspecState(t *testing.T) {
```

Comment:
Can you add comments to explain this test? Ideally link to an issue with more context.

Reply:
ok done